### ISSUE TITLE
Problem downloading Monocle3 dependencies (terra) on M1 Pro MacBook

### ISSUE THREAD

**Issue by garrett-lam** [cole-trapnell-lab/monocle3#614]
**Describe the bug**
- I am unable to download the Monocle3 dependencies ('terra') from Biocmanager

**To Reproduce**
> library(BiocManager)
Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)
> BiocManager::install(c('BiocGenerics', 'DelayedArray', 'DelayedMatrixStats',
+                        'limma', 'lme4', 'S4Vectors', 'SingleCellExperiment',
+                        'SummarizedExperiment', 'batchelor', 'Matrix.utils',
+                        'HDF5Array', 'terra', 'ggrastr'))
'getOption("repos")' replaces Bioconductor standard repositories, see '?repositories' for
details

replacement repositories:
    CRAN: https://cran.rstudio.com/

Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.1 (2022-06-23)
Installing package(s) 'terra'

  There is a binary version available but the source version is later:
      binary source needs_compilation
terra 1.5-21  1.6-3              TRUE

Do you want to install from sources the package which needs compilation? (Yes/no/cancel) yes
installing the source package ‘terra’

trying URL 'https://cran.rstudio.com/src/contrib/terra_1.6-3.tar.gz'
Content type 'application/x-gzip' length 686866 bytes (670 KB)
==================================================
downloaded 670 KB

* installing *source* package ‘terra’ ...
** package ‘terra’ successfully unpacked and MD5 sums checked
** using staged installation
configure: CC: clang -mmacosx-version-min=10.13
configure: CXX: clang++ -mmacosx-version-min=10.13 -std=gnu++11
checking for gdal-config... /opt/homebrew/bin/gdal-config
checking gdal-config usability... yes
configure: GDAL: 3.5.1
checking GDAL version >= 2.0.1... yes
checking for gcc... clang -mmacosx-version-min=10.13
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang -mmacosx-version-min=10.13 accepts -g... yes
checking for clang -mmacosx-version-min=10.13 option to accept ISO C89... none needed
checking how to run the C preprocessor... clang -mmacosx-version-min=10.13 -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking gdal.h usability... yes
checking gdal.h presence... yes
checking for gdal.h... yes
checking GDAL: linking with --libs only... no
checking GDAL: linking with --libs and --dep-libs... no
ld: warning: ignoring file /opt/homebrew/Cellar/gdal/3.5.1_1/lib/libgdal.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
Undefined symbols for architecture x86_64:
  "_GDALAllRegister", referenced from:
      _main in gdal_test-8aeace.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld: warning: directory not found for option '-L/opt/homebrew/Cellar/poppler/22.06.0_1/lib'
ld: library not found for -lcrypto
clang: error: linker command failed with exit code 1 (use -v to see invocation)
configure: Install failure: compilation and/or linkage problems.
configure: error: GDALAllRegister not found in libgdal.
ERROR: configuration failed for package ‘terra’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/terra’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/terra’

The downloaded source packages are in
	‘/private/var/folders/89/g3h_xtg92b725qqgg7b883sr0000gn/T/Rtmpl092so/downloaded_packages’
Old packages: 'edgeR', 'globals', 'MASS', 'RCurl', 'sf', 'terra'
Update all/some/none? [a/s/n]: 
n
Warning messages:
1: package(s) not installed when version(s) same as current; use `force = TRUE` to re-install:
  'BiocGenerics' 'DelayedArray' 'DelayedMatrixStats' 'limma' 'lme4' 'S4Vectors'
  'SingleCellExperiment' 'SummarizedExperiment' 'batchelor' 'Matrix.utils' 'HDF5Array' 'ggrastr' 
2: In install.packages(...) :
  installation of package ‘terra’ had non-zero exit status

**traceback()**
> traceback()
No traceback available 

**sessionInfo():**
> sessionInfo()
R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.0.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BiocManager_1.30.18

loaded via a namespace (and not attached):
[1] compiler_4.2.1 tools_4.2.1

**Comment by techgirl2022** [comment]
Hi Garrett, I also noticed this issue while trying to install monocle3 dependencies on my macbook pro and am still troubleshooting this step. Did you run into any issues while installing 'sf'?

**Comment by garrett-lam** [comment]
Yes, I have problems installing 'sf' too. I posted on their github as well.

**Comment by brookeabrown** [comment]
Hello,
I am having this issue as well. Have you had any success fixing the issue?
Thanks!

**Comment by techgirl2022** [comment]
Mac OS X was giving me issues so I tried to install monocle3 on my Windows desktop computer and that worked (the difference is only needing to install Rtools while running R version 4.1.0.
